packages/react/src/components/form/org-switching/org-switching-flow.ts

```diff
@@ -0,0 +1,458 @@
+  );
+
+  // each history entry contains a state and the event that triggered the transition to that state
+  const history: HistoryEntry[] = [{ state, event: { type: "sid_init" } }];
+
+  const { onSuccess, onError } = opts;
+
+  // notify subscribers every time the state changes
+
+  async function transition(e: Event) {
+    switch (e.type) {
+      case "sid_storeRecoveryCodes":
+        // recovery codes are only stored on register authenticator
+        if (!recoveryCodes) {
+          send({ type: "sid_login.success", user: e.user });
+          break;
+        }
+
+        setState(
+          createStoreRecoveryCodesState(send, recoveryCodes!, e.user),
+          e
+        );
+        break;
+      case "sid_login.update_context":
+        // TODO replace with a check for ready state
+        if (!logInFn || !recoverFn) break;
+
+        const updatedContext: AuthnContext = {
+          config: e.context.config,
+          options: e.context.options,
+          attempt: e.context.attempt,
+        };
+
+        setState(
+          createAuthenticatingState(
+            send,
+            updatedContext,
+            logInFn,
+            recoverFn,
+            setRecoveryCodes
+          ),
+          e
+        );
+        break;
+      case "sid_login.success":
+        // call onSuccess if present
+        if (typeof onSuccess === "function") {
+          onSuccess(e.user);
+        }
+
+        setState(createSuccessState(), e);
+        break;
+      case "sid_login.error":
+        if (state.status !== "authenticating") break;
+
+        const errorContext: ErrorState["context"] = {
+          ...state.context,
+          error: ensureError(e.error),
+        };
+
+        // call onError if present
+        if (typeof onError === "function") {
+          onError(e.error, errorContext);
+        }
+
+        setState(createErrorState(send, errorContext), e);
+        break;
+      case "sid_retry":
+        // TODO replace with a check for ready state
+        if (!logInFn || !recoverFn) break;
+
+        if (typeof cancelFn === "function") {
+          cancelFn();
+        }
+
+        const retryContext: AuthenticatingState["context"] = {
+          config: e.context.config,
+          options: e.context.options,
+          attempt: e.context.attempt + 1,
+        };
+
+        if (e.policy === "reset") {
+          setState(createInitialState(send), e);
+          break;
+        }
+
+        setState(
+          createAuthenticatingState(
+            send,
+            retryContext,
+            logInFn,
+            recoverFn,
+            setRecoveryCodes
+          ),
+          e
+        );
+        break;
+      case "sid_cancel":
+        if (typeof cancelFn === "function") {
+          cancelFn();
+        }
+
+        break;
+      default:
+        break;
+    }
+  }
+
+  // provide the flow API - interact with the flow using the state object, subscribe and unsubscribe to state changes
+  return {
```

Comment:
Most of the object we return from this factory function is the same as the other flow, so it could be a good candidate for refactor.